### ISSUE TITLE
Select component with Ecto.Enum is awkward

### ISSUE THREAD

**Issue by Carl-Foster** [phoenixframework/phoenix_live_view#2251]
LiveView version: 0.18.1
Phoenix: 1.6.7

When I was editing an existing Ecto entity that contained an Enum, I would create a select component for the Enum like this:
```heex
    <.input type="select" field={{f, :energy}} label="Energy Level">
      <:option value="" hidden={true}>Select an option</:option>
      <:option :for={value <- Ecto.Enum.values(Canvas, :energy)} value={value}>
        <%= Phoenix.Naming.humanize(value) %>
      </:option>
    </.input>
```

To render the options I have this:
```heex
        <option
          :for={opt <- @option}
          value={opt.value}
          selected={@value == opt.value}
          hidden={opt[:hidden]}
        >
          <%= render_slot(opt) %>
        </option>
```

This was correctly selecting the option on first render, but when another input was changed, the select would change to the "default" as the value converted from an atom to a string and the `selected` check was no longer true. After setting the select to another value, the `selected` check worked as expected but if changed back to the initial value, the behaviour returned.

To fix this, I had to change the `changeset` function of the Ecto schema so that the Enum fields had the `force_changes` option.

```elixir
def changeset(intial, attrs) do
  initial
  |> cast(attrs, [...])
  |> cast(attrs, [:energy], force_changes: true)
end
```

This felt very awkward, although I'm unsure if what I'm doing in general is correct or not. This is general feedback and if I'm missing something obvious please let me know.

**Comment by elliottneilclark** [comment]
This is one of the same problem that seems to be addressed by phoenixframework/phoenix/pull/5000

In that pull request @shamanime added a `attr :selected` to the `:option` slot. [Here](https://github.com/phoenixframework/phoenix/pull/5000/files#diff-95cf43325e324172c956b13560010fe8499188fe7e26c5fff76eaabda50d3e4eR258) That allows doing the equality with `to_string` when necessary (eg when using an enum). Would something like this work:

```elixir
<.input field={{form, :energy}} type="select">
  <:option value="">Select an energy type</:option>
  <:option
    :for={type <- Ecto.Enum.values(Canvas, :energy)}
    value={type}
    selected={input_value(form, :energy) == to_string(type)}
  >
    <%= Phoenix.Naming.humanize(type) %>
  </:option>
</.input>
```

 I've been testing that same solution locally as well since it allows working with embedded schemas.

**Comment by shamanime** [comment]
> This is one of the same problem that seems to be addressed by [phoenixframework/phoenix/pull/5000](https://github.com/phoenixframework/phoenix/pull/5000)
> 
> In that pull request @shamanime added a `attr :selected` to the `:option` slot. [Here](https://github.com/phoenixframework/phoenix/pull/5000/files#diff-95cf43325e324172c956b13560010fe8499188fe7e26c5fff76eaabda50d3e4eR258) That allows doing the equality with `to_string` when necessary (eg when using an enum). Would something like this work:
> 
> ```elixir
> <.input field={{form, :energy}} type="select">
>   <:option value="">Select an energy type</:option>
>   <:option
>     :for={type <- Ecto.Enum.values(Canvas, :energy)}
>     value={type}
>     selected={input_value(form, :energy) == to_string(type)}
>   >
>     <%= Phoenix.Naming.humanize(type) %>
>   </:option>
> </.input>
> ```
> 
> I've been testing that same solution locally as well since it allows for working with embedded schemas.

Nice!

I saw your comment and tested my PR with `input_value` instead of `Ecto.Changeset.get_field` but it lost the value when validation triggered or couldn't bring it back when editing.

**Comment by chrismccord** [comment]
Options are now passed as a list. Thanks!
